dietpi/dietpi-software

```diff
@@ -4332,6 +4339,223 @@ _EOF_
 			G_DIETPI-NOTIFY 0 'microblog.pub installation finished. In order to complete the setup, please run "microblog-pub configure" in a new bash session. Append the port ":8007" to the domain when being asked for it if you do not use a reverse proxy.'
 		fi
 
+		if To_Install 210 # lemmy
+		then
+			aDEPS=('make' 'gcc' 'libc6-dev' 'pkg-config' 'libssl-dev' 'protobuf-compiler' 'libprotobuf-dev' 'libpq-dev')
+
+			# This script creates the lemmy user, a dedicated postgres
+			# db in the dietpi userspace, the lemmy backend, the lemmy
+			# frontend, and apparently the pict-rs image server too
+			# (that last one is NOT clear).
+
+			## Install up-to-date rustup (cargo version in debian is too old)
+			local lemm_user='lemmy'
+			local lemm_home="/home/$lemm_user"
+			local lemm_server_path=$lemm_home/.cargo/bin
+			# User
+			##Create_User -G dialout,gpio,i2c -d "$lemm_home" "$lemm_user"
+			G_EXEC mkdir -p "$lemm_home"
+			Create_User -d "$lemm_home" "$lemm_user"
```

Comment:
Does this user require a home directory in `/home`? For service users it is common to use the software data directory instead, i.e. `/mnt/dietpi_userdata/lemmy`.

Reply:
Hmm, yeah I see. I'd prefer `/opt/lemmy` for binaries. `/home` should at best only be used for actual login users, not service users.

In this case, I actually wonder whether Rust really needs to be/should be installed as `lemmy` user or better just as `root` and installed to `/opt/lemmy`? Since we install Rust for `root` in some other cases, it might reduce some overhead, at least some disk usage. And usually it is not needed and for security reasons even better if the service user has no write access to the binary/executable, but only to a dedicated working directory for configs and data.